pyproject.toml

```diff
@@ -8,3 +8,4 @@ requires = [
 include = ["xdsl"]
 reportImportCycles = false
 reportUnnecessaryIsInstance = false
+typeCheckingMode = "strict"
```

Comment:
@math-fehr Are you ok merging this in? Some errors snuck past the reviewdog CI, since it wasn't in strict mode